pkg/operator/ceph/version/version.go

```diff
@@ -214,3 +218,39 @@ func IsInferior(a, b CephVersion) bool {
 
 	return false
 }
+
+// ValidateCephVersionsBetweenLocalAndExternalClusters makes sure an external cluster can be connected
+// by checking the external ceph versions available and comparing it with the local image provided
+func ValidateCephVersionsBetweenLocalAndExternalClusters(localVersion, externalVersion CephVersion) error {
+	logger.Debugf("local version is %s, external version is %s", localVersion.String(), externalVersion.String())
+
+	// We only support Nautilus or newer
+	if !externalVersion.IsAtLeastNautilus() {
+		return fmt.Errorf("unsupported ceph version %s, need at least nautilus, delete your cluster CR and create a new one with a correct ceph version", externalVersion.String())
+	}
+
+	// Identical version, regardless if other CRs are running, it's ok!
+	if IsIdentical(localVersion, externalVersion) {
+		return nil
+	}
+
+	// Local version must never be higher than the external one
+	if IsSuperior(localVersion, externalVersion) {
+		return fmt.Errorf("local cluster ceph version is higher %s than the external cluster %s, this must never happen", externalVersion.String(), localVersion.String())
+	}
+
+	// External cluster was updated to a minor version higher, consider updating too!
+	if localVersion.Major == externalVersion.Major {
+		if IsSuperior(externalVersion, localVersion) {
+			logger.Warningf("external cluster ceph version is a minor version higher %s than the local cluster %s, consider upgrading", externalVersion.String(), localVersion.String())
+			return nil
+		}
+	}
+
+	// The external cluster was upgraded, consider upgrading too!
+	if localVersion.Major < externalVersion.Major {
+		return fmt.Errorf("external cluster ceph version is a major version higher %s than the local cluster %s, consider upgrading", externalVersion.String(), localVersion.String())
```

Comment:
Do we really need to force the upgrade in this case? Since RADOS is compatible, even major versions are generally backward compatible. What if we log this as an error instead of blocking the orchestration? Otherwise, the local cluster would always be blocked for some short time window during an upgrade.

Reply:
Let's hope for the best then :)